sklearn/datasets/samples_generator.py

```diff
@@ -161,7 +161,7 @@ def make_classification(n_samples=100, n_features=20, n_informative=2,
         raise ValueError("Number of informative, redundant and repeated "
                          "features must sum to less than the number of total"
                          " features")
-    if 2 ** n_informative < n_classes * n_clusters_per_class:
```

Comment:
Or we could make 2 a float, or use log2 on the right hand side, or cast n_informative to a python int.

Reply:
Float 2. ** n_informative overflows at n_informative = 1024.

But we might go better with log2, since this should avoid any overflows (for sensible n_classes, n_clusters), and might be a tad faster than operations on large Python ints.

Reply:
+1 for using log2